mmcv/runner/hooks/logger/text.py

```diff
@@ -14,6 +15,12 @@ def log(self, runner):
             log_str = 'Epoch({}) [{}][{}]\t'.format(runner.mode, runner.epoch,
                                                     runner.inner_iter + 1)
         if 'time' in runner.log_buffer.output:
+            tot_time_sec = runner.log_buffer.output['time'] + \
+                runner.log_buffer.output['data_time']
+            eta_sec = tot_time_sec * \
+                (len(runner.data_loader) * runner.max_epochs - runner._iter)
+            eta_str = str(datetime.timedelta(seconds=int(eta_sec)))
+            log_str += ('eta: {}, '.format(eta_str))
```

Comment:
The time per iteration is not very stable so I think it might be better to just show a rough eta, e.g., 16.1 h

Reply:
I think the best solution is using moving average eta since runner is generalized for both long and short training.

Reply:
Yes, it would be better to use moving average or all the past time for time estimation.
Different metrics needs different processing and currently all metrics are computed as the average of past 50 iterations (time, loss, etc.) or no average (mAP, lr). An abstract of metric is required for controlling the behavior more flexibly.

Reply:
I totally agree with your opinion.